qualtran/bloqs/arithmetic/comparison.py

```diff
@@ -603,10 +612,11 @@ def build_composite_bloq(
             # We use a specially controlled Toffolli gate to implement GreaterThan.
             # If a is 1 and b is 0 then a > b and we can flip the target bit.
             ctrls = [a, b]
-            ctrls, target = bb.add(MultiControlX(cvs=(1, 0)), ctrls=ctrls, x=target)
+            for i in range(self.num_targets):
+                ctrls, targets[i] = bb.add(MultiControlX(cvs=(1, 0)), ctrls=ctrls, x=targets[i])
```

Comment:
Doing multiple Toffoli's is more expensive than doing an `And` and use a temporary ancilla to store the output; do a multi target CX controlled the result of this ancilla; and then uncompute the ancilla using `And(uncompute=True)`